recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MongoCxxConan(ConanFile):
+    name = "mongo-cxx-driver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://mongocxx.org"
+    description = "C++ Driver for MongoDB"
+    topics = ("conan", "libbsoncxx", "libmongocxx", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "arch", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**", "Find*.cmake"]
+    generators = ("cmake", "cmake_find_package")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "polyfill": ["std", "boost", "mnmlstc", "experimental"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "polyfill": "boost"
+    }
+    requires = "mongo-c-driver/1.17.2"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if self.settings.compiler == "Visual Studio" and self.options.polyfill != "boost":
+            raise ConanInvalidConfiguration("For MSVC, best to use the boost polyfill")
+
+        tools.check_min_cppstd(self, "11")
+
+        if self.options.polyfill == "std":
+            tools.check_min_cppstd(self, "17")
+
+        if self.options.polyfill == "boost":
+            self.requires("boost/1.74.0")
+
+        # Cannot model mnmlstc (not packaged, is pulled dynamically) or
+        # std::experimental (how to check availability in stdlib?) polyfill
+        # dependencies
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        if self.options.shared and not self.options["mongo-c-driver"].shared:
+            self._cmake.definitions["BUILD_SHARED_LIBS_WITH_STATIC_MONGOC"] = True
```

Comment:
```suggestion
        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
```

Seems more robust to all cases: https://github.com/mongodb/mongo-cxx-driver/blob/9ba665ecd4a92a9ff3bf72f676d5706817c311c0/CMakeLists.txt#L94